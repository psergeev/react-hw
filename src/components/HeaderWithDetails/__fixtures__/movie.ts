import { Movie } from '../../../App';

const movie: Movie = {
    poster_path: 'https://image.tmdb.org/t/p/w500/97fNAi62HawGjWru7PvVmF7RAbU.jpg',
    title: 'Kill Bill: Vol. 1',
    tagline: 'Go for the kill.',
    release_date: '2003-10-10',
    vote_average: 7.8,
    runtime: 111,
    overview: 'An assassin is shot at the altar by her ruthless employer, ' +
        'Bill and other members of their assassination circle â€“ but \'The Bride\' ' +
        'lives to plot her vengeance. Setting out for some payback, she makes a death ' +
        'list and hunts down those who wronged her, saving Bill for last.',
    genres: [
        'Action',
        'Crime',
    ],
};

export default movie;
