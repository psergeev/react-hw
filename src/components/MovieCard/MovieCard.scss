.movie-card {
    box-shadow: 0 1px 2px rgba(0, 0, 0, 0.15);
    transition: box-shadow 0.3s ease-in-out;
    cursor: pointer;
    list-style: none;
    padding: 0;
    margin: 0 15px 10px 15px;

    img {
        user-drag: none;
        user-select: none;
    }

    &:hover {
        box-shadow: 0 5px 15px rgba(0, 0, 0, 0.3);
    }

    .title-date {
        margin: 10px 0 5px 0;
        color: #7C7C7C;
        display: flex;
        justify-content: space-between;
        align-items: center;
        padding: 0 5px;

        .title {
            text-transform: uppercase;
            font-size: 16px;
            max-width: 180px;
        }

        .date {
            font-size: 14px;
            border: solid #D9D9D9 1px;
            padding: 3px 10px;
        }
    }

    .genres {
        color: #ccc;
        font-size: 12px;
        padding: 0 5px 5px 5px;
        max-width: 200px;
    }
}
