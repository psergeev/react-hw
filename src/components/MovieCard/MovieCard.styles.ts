const movieCardStyles: any = {
    card: {
        margin: '0 15px 10px 15px',
        cursor: 'pointer'
    },

    media: {
        width: '250px',
        height: '375px'
    },

    titleDate: {
        margin: '10px 0 5px 0',
        color: '#7C7C7C',
        display: 'flex',
        justifyContent: 'space-between',
        alignItems: 'center',
        padding: '0 5px',
    },

    title: {
        textTransform: 'uppercase',
        fontSize: '16px',
        maxWidth: '180px',
    },

    date: {
        fontSize: '14px',
        border: 'solid #D9D9D9 1px',
        padding: '3px 10px'
    },

    genres: {
        color: '#ccc',
        fontSize: '12px',
        padding: '0 5px 5px 5px',
        maxWidth: '200px'
    }
};

export default movieCardStyles;
