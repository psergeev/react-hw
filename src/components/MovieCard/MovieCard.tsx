import { Card, WithStyles } from '@material-ui/core';
import CardMedia from '@material-ui/core/CardMedia';
import { withStyles } from '@material-ui/core/styles';
import * as React from 'react';
import { RouteComponentProps } from 'react-router-dom';
import { Movie } from '../App/App';
import movieCardStyles from './MovieCard.styles';

interface Props extends RouteComponentProps<any> {
    handleMovieCardClick: (title: string) => void;
    movie: Movie;
}

type PropsWithStyles = Props & WithStyles<'card' | 'media' | 'titleDate' | 'title' | 'date' | 'genres'>;

function MovieCard({ handleMovieCardClick, movie, history, classes }: PropsWithStyles) {
    const _handleMovieCardClick = () => {
        history.push(`/film/${movie.id}`);
        handleMovieCardClick(movie.title);
    };

    return (
        <Card
            className={classes.card}
            onClick={_handleMovieCardClick}
        >
            <CardMedia
                className={classes.media}
                image={movie.poster_path}
                title={movie.title}
            />
            <div className={classes.titleDate}>
                <span className={classes.title}>{movie.title}</span>
                <span className={classes.date}>{movie.release_date.split('-')[0]}</span>
            </div>
            <div className={classes.genres}>{movie.genres.join(' & ')}</div>
        </Card>
    );
}

export default React.memo(withStyles(movieCardStyles)(MovieCard));
