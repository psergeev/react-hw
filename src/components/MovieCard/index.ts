import { withStyles } from '@material-ui/core/styles';
import { withRouter } from 'react-router-dom';
import MovieCard from './MovieCard';
import movieCardStyles from './MovieCard.styles';

export default withRouter(withStyles(movieCardStyles)(MovieCard));
