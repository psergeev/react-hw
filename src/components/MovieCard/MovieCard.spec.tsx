import * as React from 'react';
import * as renderer from 'react-test-renderer';
import MovieCard from './MovieCard';
import movie from './__fixtures__/movie';


test('MovieCard component works as expected', () => {
    const component = renderer.create(
        <MovieCard
            handleMovieCardClick={jest.fn()}
            movie={movie}
        />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
