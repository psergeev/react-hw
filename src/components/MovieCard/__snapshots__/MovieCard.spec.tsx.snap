// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard component works as expected 1`] = `
<div
  className="MuiPaper-root-8 MuiPaper-elevation1-11 MuiPaper-rounded-9 MuiCard-root-7 MovieCard-card-1"
  onClick={[Function]}
>
  <div
    className="MuiCardMedia-root-35 MovieCard-media-2"
    style={
      Object {
        "backgroundImage": "url(\\"https://image.tmdb.org/t/p/w500/97fNAi62HawGjWru7PvVmF7RAbU.jpg\\")",
      }
    }
    title="Kill Bill: Vol. 1"
  />
  <div
    className="MovieCard-titleDate-3"
  >
    <span
      className="MovieCard-title-4"
    >
      Kill Bill: Vol. 1
    </span>
    <span
      className="MovieCard-date-5"
    >
      2003
    </span>
  </div>
  <div
    className="MovieCard-genres-6"
  >
    Action & Crime
  </div>
</div>
`;
